gulpfile.js

```diff
@@ -94,29 +95,40 @@ gulp.task('imgmin', function () {
       .pipe(gulp.dest('public/img'));
 });
 
-// webpack build:
-gulp.task('webpack', function() {
-  webpack({
-    entry: './js/main.js',
-    output: {
-      path: __dirname + '/public/js',
-      filename: 'j.js'
-    },
-    plugins: [
-      new webpack.optimize.UglifyJsPlugin({
+// wrapper for JS build, supporting both
+// minified and unminified versions
+function jsBuild(isProduction) {
```

Comment:
This function is used below; it wraps the basic webpack behavior, but lets you specify whether you want the output minified or not. This lets you use a minified build on heroku, while using an unminified version for local development (much easier for setting breakpoints etc.).